core/tests/api_interface/images.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import PIL.Image
+from typing_extensions import assert_type
+
+import toga
+
+assert_type(toga.Image().size, tuple[int, int])
```

Comment:
Yes, if I understand correctly, this is only making assertions about the type hints themselves, but not whether those hints are actually being followed. It seems like a lot of work for no clear benefit to add hundreds of assertions that essentially say "an argument or return type hinted as X is hinted as X", but don't add any additional confidence that an X will actually be accepted or returned by the code when it's run. 

Your time might be better spent moving towards full mypy compliance in some parts of the repository, though I admit I'm ignorant about exactly what that would involve. The fact that Toga is split into semi-independent core, backend and testbed projects might make it a bit more approachable than a monolithic project like Briefcase.

In particular, a quite common class of bugs is where the public API in the core returns an object it got from the backend, but the backend doesn't comply with the public hint (e.g. #1779). It would be nice if things like that could be caught by a static type checker, rather than needing an isinstance check next to every equality check in the testbed. This would require formalizing the core / backend interface with a set of type-hinted base classes, rather than the duck-typing which is mostly used at present, which is probably a good idea even if we don't end up running mypy at all.